### DISCUSSION TITLE
Bonvision window freeze

### DISCUSSION THREAD

**Discussion by petroschalas** [bonsai-rx/bonsai#981] [Q&A]
Hello,

I am taking my first steps in bonvision and I am trying to make a looming stimulus that is activated by a button press. I managed to  design the stimulus and control it by grouping the stimulus nodes into a SelectMany node and activating it by KeyDown. However, when I reopen the file, the workflow doesn't work. The shaders window freezes when I press the key to activate the stimulus. The only way to circumvent this was to ungroup the stimulus nodes, run the workflow, and then group them again and then it would work. But sometimes when running the workflow, the stimulus would still freeze sometimes. 

Have I wired nodes in a problematic way ?

Thank you very much for your time and help. Workflow attached
[looming_stimulus.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/9166565/looming_stimulus.bonsai.zip)

**Comment by bruno-f-cruz** [comment]
Without going into the details on how you implemented the logic of your protocol, replacing all the `Timer(Reactive)` by `Timer(Shaders)` should fix the crash.

**Reply by petroschalas** [reply]
Thank you very much for your help, that worked perfectly. Do I always need to use the Timer(Shaders) when using bonvision stimuli ?

**Reply by bruno-f-cruz** [reply]
As a rule of thumb, yes.
